packages/rollup-full-node/src/app/web3-rpc-handler.ts

```diff
@@ -361,15 +361,24 @@ export class DefaultWeb3Handler
       block['transactions'] = (
         await Promise.all(
           block['transactions'].map(async (transaction) => {
-            transaction['hash'] = await this.getInternalTxHash(
-              transaction['hash']
-            )
+            transaction['hash'] = await this.getOvmTxHash(transaction['hash'])
+            let ovmTx = await this.getTransactionByHash(transaction['hash'])
+            transaction['from'] = ovmTx.from
+            transaction['to'] = ovmTx.to
+            transaction['gasLimit'] = ovmTx.gasLimit.toNumber()
```

Comment:
can you just iterate through all fields of the OVMtx and if that field exists in the internal tx, then overwrite it with the ovm field.

e.g. `to, from, v, r, s`, etc. will be overwritten, but since there is no`gasLimit` in the internal transaction, that won't be set